src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnector.java

```diff
@@ -0,0 +1,170 @@
+ */
+package io.cdap.plugin.ariba.source.connector;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.connector.BrowseDetail;
+import io.cdap.cdap.etl.api.connector.BrowseEntity;
+import io.cdap.cdap.etl.api.connector.BrowseRequest;
+import io.cdap.cdap.etl.api.connector.Connector;
+import io.cdap.cdap.etl.api.connector.ConnectorContext;
+import io.cdap.cdap.etl.api.connector.ConnectorSpec;
+import io.cdap.cdap.etl.api.connector.ConnectorSpecRequest;
+import io.cdap.cdap.etl.api.connector.PluginSpec;
+import io.cdap.cdap.etl.api.validation.ValidationException;
+import io.cdap.plugin.ariba.source.AribaBatchSource;
+import io.cdap.plugin.ariba.source.AribaServices;
+import io.cdap.plugin.ariba.source.config.AribaPluginConfig;
+import io.cdap.plugin.ariba.source.exception.AribaException;
+import io.cdap.plugin.ariba.source.metadata.AribaResponseContainer;
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import io.cdap.plugin.common.ConfigUtil;
+import okhttp3.HttpUrl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+/**
+ * AribaConnector Class
+ */
+@Plugin(type = Connector.PLUGIN_TYPE)
+@Name(ResourceConstants.PLUGIN_NAME)
+@Description("Connection to access data from Ariba.")
+public class AribaConnector implements Connector {
+  private static final Logger LOG = LoggerFactory.getLogger(AribaConnector.class);
+  private static final String METADATA_PATH = "api/analytics-reporting-view/v1";
+  private static final String ENTITY_TYPE_TEMPLATE = "template";
+  private static final String VIEW_TEMPLATE = "viewTemplateName";
+  private static final String PATH_SEGMENT = "%s/viewTemplates";
+  private static final String RECORDS = "Records";
+  private static final String TOKEN = "PageToken";
+  private static final Gson gson = new Gson();
+  private final AribaConnectorConfig config;
+  private String accessToken;
+
+  public AribaConnector(AribaConnectorConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void test(ConnectorContext connectorContext) throws ValidationException {
+    FailureCollector collector = connectorContext.getFailureCollector();
+    config.validateCredentials(collector);
+    try {
+      config.validateToken(collector);
+    } catch (AribaException | IOException e) {
+      LOG.error(" Unable to validate token", e);
```

Comment:
Add failure to collector

Reply:
failures are already added in the collector in the inner validate methods.

Reply:
you shouldn't be throwing exceptions in validateToken(). Catch any exceptions, an add a failure to the collector.
In your current implementation you're ignoring exceptions, which is incorrect